### ISSUE TITLE
BUG: Hunks Extension Memory Issues

### ISSUE THREAD

**Issue by philip-iii** [SoftwareIntrospectionLab/MininGit#37]
The Hunks extension suffers from memory issues, specifically, lines 264-269:

```
    # Get the patches from this repository
    query = "select p.commit_id, p.patch, s.rev from patches p, scmlog s " + \
            "where p.commit_id = s.id and " + \
            "s.repository_id = ? and " + \
            "p.patch is not NULL"
    read_cursor.execute(statement(query, db.place_holder),(repo_id,))
```

tend to fail with larger amounts of patches. Performing the SQL query on a 40000+ patches (700+ MB of content) results in a crash. My suggestion would be to segment the query in smaller chunks and perform the whole operation incrementally. Manual segmentation of 10000 entries at a time worked fine on my setup. 

Tried the same query in a Java-based setting and was able to perform the query with 24000 entries at a time (with increased memory settings).

I'm guessing this depends largely on the size of the patches, but a setting of 1000 should probably run without any issues. Making this configurable may also be helpful

**Comment by cflewis** [comment]
Thanks for filing this one, I noticed it too but forgot to add the ticket.

Agree with your resolution strategy.

**Comment by cflewis** [comment]
Altered Hunks to get 100 patches at a time.

Closed by 15c151996f2dafbaeaa7a53a6b673c6a9c6848a7, Closed by 15c151996f2dafbaeaa7a53a6b673c6a9c6848a7.
